infrastructure/collections/src/main/java/tech/pegasys/teku/infrastructure/collections/cache/LRUCache.java

```diff
@@ -28,31 +28,27 @@
  */
 public class LRUCache<K, V> implements Cache<K, V> {
 
+  public static <K, V> LRUCache<K, V> create(int capacity) {
+    return new LRUCache<>(
+        SynchronizedLimitedMap.createEmpty(),
+        initEntries ->
+            SynchronizedLimitedMap.create(
+                capacity, LimitStrategy.DROP_OLDEST_ELEMENT, initEntries));
+  }
+
+  private final Function<Map<K, V>, Map<K, V>> backStorageCopier;
   private final Map<K, V> cacheData;
-  private final int maxCapacity;
 
-  /**
-   * Creates cache
-   *
-   * @param capacity Size of the cache
-   */
-  public LRUCache(int capacity) {
-    this(capacity, Collections.emptyMap());
-  }
+  private LRUCache(
+      Map<K, V> initialCachedContent, Function<Map<K, V>, Map<K, V>> backStorageCopier) {
 
-  private LRUCache(int capacity, Map<K, V> initialCachedContent) {
-    this.maxCapacity = capacity;
-    Map<K, V> cacheMap = LimitedMap.create(maxCapacity);
-    // copy safely, initialCachedContent is always a SynchronizedMap instance
-    synchronized (initialCachedContent) {
-      cacheMap.putAll(initialCachedContent);
-    }
-    this.cacheData = cacheMap;
+    this.backStorageCopier = backStorageCopier;
+    this.cacheData = backStorageCopier.apply(initialCachedContent);
```

Comment:
I'm unclear why this is being passed in as a function when it's always the same lambda supplied.  Seems like it would make more sense to just inline it.  I think it then becomes just:

```suggestion
    this.cacheData = SynchronizedLimitedMap.create(
                 capacity, LimitStrategy.DROP_OLDEST_ELEMENT, initialCachedContent);
```